src/azul/chalice.py

```diff
@@ -189,15 +193,37 @@ def _api_gateway_context_middleware(self, event, get_response):
         finally:
             config.lambda_is_handling_api_gateway_request = False
 
+    hsts_max_age = 60 * 60 * 24 * 365 * 2
+
+    # Headers added to every response from an AWS Lambda function. Use of these
+    # headers prevent known security vulnerabilities.
+    #
+    security_headers = {
+        'Referrer-Policy': 'strict-origin-when-cross-origin',
+        'Strict-Transport-Security': jw(f'max-age={hsts_max_age};',
+                                        'includeSubDomains;',
+                                        'preload'),
+        'X-Content-Type-Options': 'nosniff',
+        'X-Frame-Options': 'DENY',
+        'X-XSS-Protection': '1; mode=block'
+    }
+
+    # Headers added to 4XX and 5XX responses from the API Gateway. This includes
+    # the security headers that would be added to AWS Lambda function responses,
+    # along with a CSP that is suitable for an error page, but not in other
+    # cases (e.g. the Swagger UI).
+    #
+    api_gateway_4xx_5xx_headers = {
+        **security_headers,
+        'Content-Security-Policy': jw('default-src', sq('self')),
```

Comment:
I assume this CSP is suitable to all pages, except the Swagger login stuff. You should code it that way. Move the CSP header to `security_headers` and add an exception to drop the header where it is problematic. Prefix the exception with a FIXME referring to the issue that adds the nonce CSP.